pycbc/distributions/sampling_from_config.py

```diff
@@ -0,0 +1,100 @@
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; If not, see <http://www.gnu.org/licenses/>.
+
+
+import argparse
+import numpy as np
+from pycbc.workflow import WorkflowConfigParser, configuration
+import pycbc.distributions as distributions
+from pycbc.coordinates import spherical_to_cartesian
+from pycbc.cosmology import redshift_from_comoving_volume, distance_from_comoving_volume
+
+
+def draw_samples_from_config(config_path, samples_num=1, seed=150914, **kwds):
+    """ Generate sampling points from a standlone .ini file.
+    Parameters
+    ----------
+    config_path : str
+        The path of the .ini file.
+    samples_num : int
+        The number of samples.
+    seed: int
+        The random seed for sampling.
+    Returns
+    --------
+    samples : list
+        The parameter values [{param:value}] of sampling points in the parameter space.
+    """
+
+    np.random.seed(seed)
+    parser = argparse.ArgumentParser(description=__doc__,
+                formatter_class=argparse.RawDescriptionHelpFormatter)
+    configuration.add_workflow_command_line_group(parser)
+    parser.add_argument('--dist-section', default='prior',
+                        help='What section in the config file to load '
+                            'distributions from. Default is prior.')
+    parser.add_argument('--variable-params-section', default='variable_params',
+                        help='What section in the config file to load the '
+                            'parameters to vary. Default is variable_params.')
+    parser.add_argument('--static-params-section', default="static_params",
+                        help='What section to load the static params from. Default '
+                            'is static_params.')
+    opts = parser.parse_args()
+    # Add .ini file path.
+    opts.config_files = [config_path]
+    cp = WorkflowConfigParser.from_cli(opts)
+
+    # Get the vairable and static arguments from the .ini file.
+    variable_params, static_params = distributions.read_params_from_config(cp,
+        prior_section=opts.dist_section,
+        vargs_section=opts.variable_params_section,
+        sargs_section=opts.static_params_section)
+    constraints = distributions.read_constraints_from_config(cp)
+
+    # Get prior distribution for each variable parameter.
+    dists = distributions.read_distributions_from_config(cp, opts.dist_section)
+
+    # Construct class that will draw the samples.
+    randomsampler = distributions.JointDistribution(variable_params, *dists,
+                                **{"constraints" : constraints})
+
+    # Draw samples from prior distribution.
+    samples = randomsampler.rvs(size=int(samples_num))
+
+    # Pair the value with the parameter name.
+    samples_list = []
```

Comment:
Why do you need the following? This seems very hard coded for specific parameters. You don't know if those will be used by someone or not. We have the ability to specify such transforms in the config file direclty, so perhaps use that machinery to interpret generic configuration-given transformations. Then you just return the samples structured array.

Reply:
See https://github.com/gwastro/pycbc/blob/87ce2d5d5d491e8629b62b99f95a9299650c2d60/bin/pycbc_create_injections#L195

(you can also see the example config file in the docs of the same exe for something to test with if you like)